@here/harp-geoutils/lib/coordinates/GeoPolygon.ts

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Vector2 } from "three";
+
+import { GeoBox } from "./GeoBox";
+import { GeoCoordinates, MAX_LATITUDE, MIN_LATITUDE } from "./GeoCoordinates";
+import { GeoCoordinatesLike } from "./GeoCoordinatesLike";
+import { GeoPolygonLike } from "./GeoPolygonLike";
+
+type MinThreeItemsArray<T> = [T, T, T, ...T[]];
+
+export type GeoPolygonCoordinates = MinThreeItemsArray<GeoCoordinatesLike | GeoCoordinates>;
+
+/**
+ * A GeoPolygon in 2D Space (altitudes will be ignored).
+ * Coordinates are expected in counter-clockwise order, for concave polygons a sorting is
+ * available.
+ * Clockwise ordered or selfintersecting Polygons might lead to no or unexpected results.
+ *
+ * @beta @internal
+ */
+export class GeoPolygon implements GeoPolygonLike {
+    coordinates: GeoPolygonCoordinates;
+    private readonly m_center: GeoCoordinates | undefined;
+
+    /**
+     * Creates a GeoPolygon instance
+     *
+     * @param coordinates An array of GeoCoordinates acting as the Vertices of the Polygon.
+     * @param center Adds a user defiened center coordinate
```

Comment:
I think center should not go into GeoPolygon. If there is an API that also needs a center it should have a separate center parameter

Reply:
ok, done